openedx/core/lib/block_structure/transformer.py

```diff
@@ -125,4 +126,55 @@ def transform(self, usage_info, block_structure):
                 block structure, with already collected data for the
                 transformer, that is to be transformed in place.
         """
-        pass
+        raise NotImplementedError
+
+
+class OptimizedTransformer(BlockStructureTransformer):
```

Comment:
Given this discussion - I'm now questioning the design pattern of creating a new subclass for this.  Every time we decide to add a new method on a Transformer are we going to create a new class?

I'd rather make it easier for Transformer developers and also avoid complex class hierarchies.  I prefer the pythonic way of thinking about this.

So:
* Can we just have one BlockStructureTransformer class?
* Make it an abstract class using the `abc.ABCMeta` class.
* Declare both `transform` and `transform_block_filter` on the abstract class and have them raise `NotImplementedError`.
* Resurrect the `_is_block_filter_implemented` method on `BlockStructureTransformers` to determine whether or not the transformer supports the filter method.